### PR TITLE
feat: make polling input generic

### PR DESCRIPTION
**Pull request by bergeron**
## Explanation

Allow the polling controllers to accept generic input when starting a poll.

Today they accept:
```
networkClientId: NetworkClientId,
options: Json
```

But controllers may want unique polling loops over other data.

This PR allows controllers to use any serializable type as their polling input.  Controllers pass their input type as a generic argument to the base class, making it more type safe than then previous `options: Json`.

An example of how this would be used in the future, is how `CurrencyRateController` only needs a polling loop for each unique native currency.  So it would define:

```
type CurrencyRatePollingInput = {
  nativeCurrency: string;
};
```

And you would initiate polling with:
```
const currencyRateController = new(...);
const token1 = currencyRateController.startPolling({nativeCurrency: 'ETH'});
const token2 = currencyRateController.startPolling({nativeCurrency: 'POL'});
```

### `@metamask/polling-controller`

- **BREAKING**: The input to start polling is now a generic type that can be any input, instead of requiring a network client id like before.  The functions interfaces are updated to accommodate this.  `startPollingByNetworkClientId` is now `startPolling`.  And `onPollingComplete` now returns the entire input object, instead of a network client id.

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [ ] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

File: packages/polling-controller/src/AbstractPollingController.ts [hunk 1]
```diff
@@ -24,45 +18,35 @@ export const getKey = (
  */
 // TODO: Either fix this lint violation or explain why it's necessary to ignore.
 // eslint-disable-next-line @typescript-eslint/naming-convention
-export function AbstractPollingControllerBaseMixin<TBase extends Constructor>(
-  Base: TBase,
-) {
+export function AbstractPollingControllerBaseMixin<
+  TBase extends Constructor,
+  PollingInput,
```
Comment (jiexi) [inline packages/polling-controller/src/AbstractPollingController.ts:23]: does this need to `extends Json` since we need it to be stringifiable?

Comment (bergeron) [reply packages/polling-controller/src/AbstractPollingController.ts:23]: I was also wondering this.  We do want the input to be serializable, but i wasn't sure what constraints would be necessary to guarantee that.

**Review by jiexi — APPROVED** [review]
clean and beautiful change. Thank you!

**Review by adonesky1 — APPROVED** [review]
Looks **Great** to Me!

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**